lxd/storage_zfs.go

```diff
@@ -547,7 +547,73 @@ func (s *storageZfs) StoragePoolUpdate(writable *api.StoragePoolPut, changedConf
 	}
 
 	if shared.StringInSlice("volume.size", changedConfig) {
-		return fmt.Errorf("the \"volume.size\" property cannot be changed")
+		volumes, err := db.StoragePoolVolumesGet(s.d.db, s.poolID,
+			[]int{storagePoolVolumeTypeContainer})
+		if err != nil {
+			if err != db.NoSuchObjectError {
+				return err
+			}
+		}
+
+		// The name of a container is always identical to the name of
+		// its storage volume.
+		for _, volume := range volumes {
+			ct, err := containerLoadByName(s.d, volume.Name)
+			if err != nil {
+				return err
+			}
+
+			// Quota do not apply to snapshots (I *think*.)
+			if ct.IsSnapshot() {
+				continue
+			}
+
+			// Check if the storage volume itself already has local
+			// "size" property set.
+			if volume.Config["size"] != "" {
+				continue
+			}
+
+			// Check if the container has a "size" property set on
+			// its root disk device in its local devices.
+			localDevices := ct.LocalDevices()
+			_, rootDiskDevice, _ := containerGetRootDiskDevice(localDevices)
+			// Check if the container inherits a "size" property
+			// from a root disk device in one of its profiles.
+			if rootDiskDevice["size"] == "" {
+				expandedDevices := ct.ExpandedDevices()
+				_, rootDiskDevice, _ = containerGetRootDiskDevice(expandedDevices)
+			}
```

Comment:
Why that complicated logic? Just always look in expandedDevices. If there's a container-local root device, that'll find it.